src/main/java/com/iggroup/universityworkshopmw/domain/services/OpenPositionsService.java

```diff
@@ -0,0 +1,88 @@
+package com.iggroup.universityworkshopmw.domain.services;
+
+import com.iggroup.universityworkshopmw.domain.exceptions.NoAvailableDataException;
+import com.iggroup.universityworkshopmw.domain.model.OpenPosition;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+@Slf4j
+@Component
+public class OpenPositionsService {
+   private Map<String, List<OpenPosition>> clientPositionStore;
+
+   public OpenPositionsService() {
+      clientPositionStore = new HashMap<>();
+   }
+
+   public List<OpenPosition> getOpenPositionsForClient(String clientId) throws NoAvailableDataException {
+      List<OpenPosition> openPositions = clientPositionStore.get(clientId);
+
+      if (openPositions != null) {
+         return openPositions;
+      }
+      throw new NoAvailableDataException("No open positions exist for client: " + clientId);
+   }
+
+   public void addOpenPositionForClient(String clientId, OpenPosition openPosition) {
+      List<OpenPosition> openPositions = clientPositionStore.get(clientId);
+      if (openPositions != null) {
+         openPositions.add(openPosition);
+      } else {
+         clientPositionStore.put(clientId, newArrayList(openPosition));
+      }
```

Comment:
We need to also update client funds, by subtracting the openPosition.openingPrice * buy size ?